testsuite/smoke-test/src/upgrade.rs

```diff
@@ -159,11 +120,9 @@ async fn test_upgrade_flow() {
             .unwrap()
             .status
             .success());
-    }
 
-    // Two transactions has been emitted to root account.
-    *env.aptos_public_info().root_account().sequence_number_mut() +=
-        1 + package_path_list.len() as u64;
+        *env.aptos_public_info().root_account().sequence_number_mut() += 1;
+    }
 
     // Test the module publishing workflow
     let base_dir = std::path::Path::new(env!("CARGO_MANIFEST_DIR"));
```

Comment:
Can be in a followup PR: Can you check that the flags are turned on, gas schedule is updated, etc.?